applications/GeoMechanicsApplication/tests/cpp_tests/test_U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -0,0 +1,264 @@
+            r_model_part.AddNodalSolutionStepVariable(WATER_PRESSURE);
+
+            // set up condition
+            ModelPart::ConditionType::Pointer p_cond = SetUpUPwLysmerAbsorbingCondition2D2NCondition(r_model_part);
+            const auto& r_process_info = r_model_part.GetProcessInfo();
+
+            // add dofs to nodes
+            for (auto& node : r_model_part.Nodes())
+            {
+                node.AddDof(DISPLACEMENT_X);
+                node.AddDof(DISPLACEMENT_Y);
+                node.AddDof(WATER_PRESSURE);
+            }
+
+            p_cond->GetGeometry()[0].FastGetSolutionStepValue(DISPLACEMENT_X) = 1.0;
+            p_cond->GetGeometry()[0].FastGetSolutionStepValue(DISPLACEMENT_Y) = 2.0;
+
+            p_cond->GetGeometry()[1].FastGetSolutionStepValue(DISPLACEMENT_X) = 3.0;
+            p_cond->GetGeometry()[1].FastGetSolutionStepValue(DISPLACEMENT_Y) = 4.0;
+
+
+            // Perfrom test, calculate left hand side
+            size_t condition_size = 6;
+            Matrix rLeftHandSideMatrix = ZeroMatrix(condition_size, condition_size);
+            Vector right_hand_side_vector = ZeroVector(condition_size);
+
+            //const auto& r_process_info = r_model_part.GetProcessInfo();
+            p_cond->CalculateLocalSystem(rLeftHandSideMatrix, right_hand_side_vector, r_process_info);
+
+            // set expected_results
+            Matrix expected_matrix = ZeroMatrix(condition_size, condition_size);
+            expected_matrix(0, 0) = 5.0 / 100 / 4;
+            expected_matrix(0, 2) = expected_matrix(0, 0);
+            expected_matrix(2, 0) = expected_matrix(0, 0);
+            expected_matrix(2, 2) = expected_matrix(0, 0);
+
+            expected_matrix(1, 1) = 4.0 / 100 / 4;
+            expected_matrix(1, 3) = expected_matrix(1, 1);
+            expected_matrix(3, 1) = expected_matrix(1, 1);
+            expected_matrix(3, 3) = expected_matrix(1, 1);
+
+            Vector expected_rhs = ZeroVector(condition_size);
+            expected_rhs[0] = -(expected_matrix(0, 0) * 1.0 + expected_matrix(0, 2) * 3.0);
+            expected_rhs[1] = -(expected_matrix(1, 1) * 2.0 + expected_matrix(1, 3) * 4.0);
+            expected_rhs[2] = -(expected_matrix(2, 0) * 1.0 + expected_matrix(2, 2) * 3.0);
+            expected_rhs[3] = -(expected_matrix(3, 1) * 2.0 + expected_matrix(3, 3) * 4.0);
+
+            // compare results
+            for (unsigned int i = 0; i < condition_size; ++i)
+            {
+                // compare lhs
+                for (unsigned int j = 0; j < condition_size; ++j)
+                {
+                    KRATOS_EXPECT_NEAR(
+                        rLeftHandSideMatrix(i, j),
+                        expected_matrix(i, j),
+                        1.0e-6);
+                }
+                // compare rhs
+                KRATOS_EXPECT_NEAR(
+                    right_hand_side_vector[i],
+                    expected_rhs[i],
+                    1.0e-6);
+            }
+        }
```

Comment:
I expect we should be able to re-use the matrix and vector checks here, so this should be equivalent:
```suggestion
  KRATOS_EXPECT_MATRIX_NEAR(rLeftHandSideMatrix, expected_matrix(i, j), 1.0e-6);
  KRATOS_EXPECT_VECTOR_NEAR(right_hand_side_vector, expected_rhs, 1.0e-6);
```